### PR TITLE
upgrade odb/codegen

### PR DESCRIPTION
**Pull request by mpollmeier**
* most (but not all) changes related to https://github.com/ShiftLeftSecurity/overflowdb-codegen/pull/85
* please carefully review the default values I provided here! wasn't one of those integer values supposed to be 
  * especially: order: looks like it used to default to -1? See change in TypeDeclest

### CODE REVIEW

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Base.scala [hunk 1]
```diff
@@ -45,84 +45,84 @@ object Base extends SchemaBase {
...
         name = "IS_EXTERNAL",
-        valueType = ValueTypes.BOOLEAN,
-        cardinality = Cardinality.One,
+        valueType = ValueType.Boolean,
         comment = """Indicates that the construct (METHOD or TYPE_DECL) is external, that is,
                     |it is referenced but not defined in the code (applies both to insular
                     |parsing and to library functions where we have header files only)
                     |""".stripMargin
       )
+      .mandatory(PropertyDefaults.Boolean)
```
Comment (bbrehm) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Base.scala:63]: Should be explicit `false`

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala [hunk 2]
```diff
@@ -35,13 +37,13 @@ object CallGraph extends SchemaBase {
                     |ignored. It is suggested to set it to -1.
                     |""".stripMargin
       )
+      .mandatory(PropertyDefaults.Int)
```
Comment (bbrehm) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala:40]: I think -2 would be more sensible -- but the docstring suggests -1 when this is not applicable; both need to be consistent.

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala [hunk 3]
```diff
@@ -115,15 +117,15 @@ object CallGraph extends SchemaBase {
...
-        cardinality = Cardinality.One,
+        valueType = ValueType.String,
         comment = """This field holds the dispatch type of a call, which is either `STATIC_DISPATCH` or
             |`DYNAMIC_DISPATCH`. For statically dispatched method calls, the call target is known
             |at compile time while for dynamically dispatched calls, it can only be determined at
             |runtime as it may depend on the type of an object (as is the case for virtual method
             |calls) or calculation of an offset.
             |""".stripMargin
       )
+      .mandatory(PropertyDefaults.String)
```
Comment (bbrehm) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala:128]: We could consider either STATIC_DISPATCH or DYNAMIC_DISPATCH as default

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala [hunk 4]
```diff
@@ -38,6 +38,7 @@ object Method extends SchemaBase {
                     |implementor to fit the source language.
                     |""".stripMargin
       )
+      .mandatory(PropertyDefaults.String)
```
Comment (bbrehm) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala:41]: We could consider empty-string for this one

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala [hunk 5]
```diff
@@ -75,14 +76,14 @@ object Method extends SchemaBase {
...
-        valueType = ValueTypes.BOOLEAN,
-        cardinality = Cardinality.One,
+        valueType = ValueType.Boolean,
         comment = """
                     |Specifies whether a parameter is the variadic argument handling parameter of
                     |a variadic method. Only one parameter of a method is allowed to have this
                     |property set to true.
                     |""".stripMargin
       )
+      .mandatory(PropertyDefaults.Boolean)
```
Comment (bbrehm) [inline schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala:86]: Explicit `false`

**Review by bbrehm — COMMENTED** [review]
Some ideas -- but don't change it until the others chimed in.

Are we able to set defaults on a per-node basis?

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala [hunk 6]
```diff
@@ -52,13 +55,13 @@ object Ast extends SchemaBase {
...
         name = "ORDER",
-        valueType = ValueTypes.INTEGER,
-        cardinality = Cardinality.One,
+        valueType = ValueType.Int,
         comment = """This integer indicates the position of the node among
             |its siblings in the AST. The left-most child has an
             |order of 0.
             |""".stripMargin
       )
+      .mandatory(PropertyDefaults.Int)
```
[the remark this answers is not in the document]
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala:64]: spoke to markus and fabs and they also suggested -1 for all integer number types. updated the defaults.

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala (hunk 3 above)
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala:128]: @fabsx00 @ml86 wdyt?

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala (hunk 5 above)
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala:86]: done

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala (hunk 4 above)
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Method.scala:41]: I agree, changed.

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala (hunk 2 above)
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala]: i changed it to explicit `-1` as per the docstring - @fabsx00 @ml86 wdyt?

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Base.scala (hunk 1 above)
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Base.scala]: done

**Comment by mpollmeier** [comment]
thanks for carefully checking the default values @bbrehm! I've made a few changes
@fabsx00 @ml86 your feedback is much needed here, those defaults are rather important.

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala (hunk 3 above)
Comment (fabsx00) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala:128]: Hard to say. There are languages such as C, where STATIC_DISPATCH would be a good default, and scripting languages where DYNAMIC_DISPATCH is. So, I think I'd go with not giving this a default.

**Review by fabsx00 — APPROVED** [review]

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala (hunk 6 above)
[the remark this answers is not in the document]
Comment (ml86) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala:64]: I would suggest putting a -1 explicitly as you did for ARGUMENT_INDEX.

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala (hunk 3 above)
Comment (ml86) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala:128]: I think we are fine with the default default.

**Review by ml86 — APPROVED** [review]

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala (hunk 3 above)
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/CallGraph.scala:128]: :+1:

File: schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala (hunk 6 above)
[the remark this answers is not in the document]
Comment (mpollmeier) [reply schema/src/main/scala/io/shiftleft/codepropertygraph/schema/Ast.scala]: done

**Pull request merged**